File: SSSegmentedViewExample/ViewController.m
//
//  ViewController.m
//  SSSegmentedViewExample
//
//  Created by xiaobei on 2017/7/12.
//  Copyright © 2017年 xiaobei. All rights reserved.
//

#import "ViewController.h"
#import "SSTestViewController.h"
#import "SSSegmentedView.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [self setUI];
}

- (void)setUI{
    NSInteger count = 8;
    NSMutableArray *titleArray = [NSMutableArray array];
    for(int i=0; i<count; i++){
        [titleArray addObject:[NSString stringWithFormat:@"按钮%d",i]];
    }
    NSMutableArray *viewArray = [NSMutableArray array];
    for(int i=0; i<count; i++){
        UIViewController *VC = [[SSTestViewController alloc] init];
        // 这里讲控制器的view添加到当前控制器里
        [self addChildViewController:VC];
        [viewArray addObject:VC.view];
    }
    
    SSSegmentedView *segmentedView = [[SSSegmentedView alloc] initWithFrame:CGRectMake(0, 100, 320, 320) titleArray:titleArray viewArray:viewArray];
    [self.view addSubview:segmentedView];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: SSSegmentedViewExample/SSSegmentedViewExample/ViewController.swift
//
//  ViewController.swift
//  SSSegmentedViewExample
//
//  Created by xiaobei on 2017/8/23.
//  Copyright © 2017年 xiaobei. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        var contentViews = [UIView]()
        let titles = ["标题1", "标题11", "标题111", "标题1111", "标题11111"]
        for _ in 0..<titles.count {
            let vc = testViewController()
            addChildViewController(vc)
            contentViews.append(vc.view)
        }
        
        let frame = CGRect(x: 0, y: 20, width: view.bounds.width, height: view.bounds.height)
        let segmentView = SSSegmentedView(frame: frame, titles: titles, contentViews: contentViews)
        segmentView.backgroundColor = UIColor.white
        segmentView.sliderStyle = .center
        segmentView.isShowBottomLine = false
        segmentView.viewIndex = { index in
            print("----滚动位置\(index)")
        }
        
        view.addSubview(segmentView)
        segmentView.selectIndex = 2
        segmentView.topHeight = 100
    }

}

